### PR TITLE
[patch] 2024.6.4

### PR DESCRIPTION
**Pull request by danh91**
### Feature

- feat: Add field to unified class for future support of message level

### Chore

- chore: Add tests for `UPS` EU rates parsing
- chore: make ups upper case for uniform charges details
- chore: use localhost address for default `NEXT_PUBLIC_KARRIO_PUBLIC_URL` during build

### Refactor

- refactor: #673

### CODE REVIEW

File: modules/connectors/ups/karrio/providers/ups/rate.py [hunk 1]
```diff
@@ -27,35 +28,35 @@ def _extract_details(
...
                     provider_units.SurchargeType.map(
                         str(getattr(c, "Code", None) or getattr(c, "Type", None))
-                    ).name.capitalize().replace("_", " ")
-                    if provider_units.SurchargeType.map(
-                        str(getattr(c, "Code", None) or getattr(c, "Type", None))
-                    ).name
-                    else (getattr(c, "Code", None) or getattr(c, "Type", None))
+                    )
+                    .name_or_key.replace("_", " ")
+                    .upper()
```
Comment (jacobshilitz) [inline modules/connectors/ups/karrio/providers/ups/rate.py:57]: I was using `capitalize` so it matches `Base charge`

Comment (jacobshilitz) [reply modules/connectors/ups/karrio/providers/ups/rate.py:57]: also i see that `upper` has `.replace("_", " ")` already in it

Comment (danh91) [reply modules/connectors/ups/karrio/providers/ups/rate.py:57]: The issue was that things like HST would ended up being `Hst` so to make it uniform that I replace to upper.
I will also make `Base Charge` upper to keep all uniform.

Comment (jacobshilitz) [reply modules/connectors/ups/karrio/providers/ups/rate.py:57]: > The issue was that things like HST would ended up being Hst so to make it uniform that I replace to upper.
I will also make Base Charge upper to keep all uniform.

correct that was the reason i wrote it so complicated... :)

Comment (danh91) [reply modules/connectors/ups/karrio/providers/ups/rate.py:57]: tested without `.replace("_", " ")` and just upper doesn't seem to remove them :)

File: modules/connectors/canadapost/karrio/providers/canadapost/shipment/create.py [hunk 2]
```diff
@@ -89,8 +90,8 @@ def shipment_request(
     label_encoding, label_format = provider_units.LabelType.map(
         payload.label_type or "PDF_4x6"
     ).value
-    group_id = str(uuid.uuid4().hex)
-    customer_request_ids = [f"{group_id}" for _ in range(len(packages))]
+    group_id = lib.to_date(datetime.datetime.now(), "%Y%m%d")
+    customer_request_ids = [f"{group_id}{_}" for _ in range(len(packages))]
```
Comment (jacobshilitz) [inline modules/connectors/canadapost/karrio/providers/canadapost/shipment/create.py:94]:
```suggestion
    customer_request_id = str(uuid.uuid4().hex)
    customer_request_ids = [f"{customer_request_id}{_}" for _ in range(len(packages))]
```
if `group_id` is not unique then we need to generate a special uuid for it

File: modules/connectors/ups/karrio/providers/ups/rate.py (hunk 1 above)
Comment (jacobshilitz) [reply modules/connectors/ups/karrio/providers/ups/rate.py:57]: is it this the method? when i click on github it showed this, maybe they are wrong https://github.com/karrioapi/karrio/blob/5b235d94ca02495a7d3996425653ac9b4b10b56c/modules/core/karrio/server/core/utils.py#L194-L198

File: modules/connectors/canadapost/karrio/providers/canadapost/shipment/create.py (hunk 2 above)
Comment (danh91) [reply modules/connectors/canadapost/karrio/providers/canadapost/shipment/create.py:94]: True forgot to update this one.

**Pull request merged**